proposals/lip-0042.md

```diff
@@ -135,11 +139,11 @@ This function is used to retrieve the reward of a block at a given height.
 ##### Execution
 
 ```python
-def getBlockReward(blockHeader: BlockHeader) -> tuple[uint64, uint32]:
+def getBlockReward(blockHeader: BlockHeader, blockAssets: BlockAssets) -> tuple[uint64, uint32]:
   if not Token.userSubstoreExists(blockHeader.generatorAddress, TOKEN_ID_REWARD):
     return (0, REWARD_REDUCTION_NO_ACCOUNT)
 
-  if Random.isSeedRevealValid(blockHeader.generatorAddress, blockHeader.seedReveal) == False:
+  if Random.isSeedRevealValid(blockHeader.generatorAddress, blockAssets) == False:
     return (0, REWARD_REDUCTION_SEED_REVEAL)
 
   defaultReward = getDefaultRewardAtHeight(blockHeader.height)
```

Comment:
minor suggestion: Should we use // for integer division in line 151 and remove line 156?

Reply:
Yes, thanks. Updated.